solr/core/src/java/org/apache/solr/highlight/UnifiedSolrHighlighter.java

```diff
@@ -456,5 +466,47 @@ protected Predicate<String> getFieldMatcher(String field) {
 
       return NOT_REQUIRED_FIELD_MATCH_PREDICATE;
     }
+
+    protected FieldHighlighter getFieldHighlighter(
+        String field, Query query, Set<Term> allTerms, int maxPassages) {
+
+      if (!"stripHTML".equals(params.getFieldParam(field, HighlightParams.ENCODER, "simple"))) {
+        return super.getFieldHighlighter(field, query, allTerms, maxPassages);
+      }
+
+      UHComponents components = getHighlightComponents(field, query, allTerms);
+      OffsetSource offsetSource = getOptimizedOffsetSource(components);
+
+      return new FieldHighlighter(
+          field,
+          getOffsetStrategy(offsetSource, components),
+          new SplittingBreakIterator(getBreakIterator(field), UnifiedHighlighter.MULTIVAL_SEP_CHAR),
+          getScorer(field),
+          maxPassages,
+          getMaxNoHighlightPassages(field),
+          getFormatter(field)) {
```

Comment:
If there was a `newFieldHighlighter` method as proposed in https://github.com/apache/lucene/pull/821 then the various `get...` calls here would not need duplicating when extending `FieldHighlighter` like this.